app/README-app.md

```diff
@@ -1,19 +1,110 @@
+
+### Privileges to objects
+
+The application will request access to:
+- Monte Carlo backend services at `https://artemis.getmontecarlo.com`.
+- Stored Procedure deployed as part of the setup process in the helper database, see steps [here](#setup-code).
+
+Additionally, for each monitored database, you'll need to execute [this script](https://raw.githubusercontent.com/monte-carlo-data/sna-artemis-agent/refs/heads/main/scripts/permissions.sql), which grants the following privileges to the role `MCD_AGENT_ROLE`:
+
+`IMPORTED PRIVILEGES` on **DATABASE**
+To access query history.
+
+`USAGE`,`MONITOR`on **DATABASE**
+To access metadata.
+
+`USAGE`,`MONITOR`on **ALL SCHEMAS IN DATABASE**
+To access metadata.
+
+`REFERENCES` on **ALL TABLES IN DATABASE**, **ALL VIEWS IN DATABASE**, **ALL EXTERNAL TABLES IN DATABASE**
+To access metadata.
+
+`MONITOR` on **ALL DYNAMIC TABLES IN DATABASE**
+To access metadata.
+
+`REFERENCES` on **FUTURE TABLES IN DATABASE**, **FUTURE VIEWS IN DATABASE**, **FUTURE EXTERNAL TABLES IN DATABASE**
+To access metadata for future objects.
+
+`MONITOR` on **FUTURE DYNAMIC TABLES IN DATABASE**
+To access metadata for future dynamic tables.
+
+`USAGE`,`MONITOR`on **FUTURE SCHEMAS IN DATABASE**
+To access metadata for future schemas.
+
+`SELECT` on **ALL TABLES IN DATABASE**, **ALL VIEWS IN DATABASE**, **ALL EXTERNAL TABLES IN DATABASE**, **ALL DYNAMIC TABLES IN DATABASE**, **ALL STREAMS IN DATABASE**
+Read-only access to query objects.
+
+`SELECT` on **FUTURE TABLES IN DATABASE**, **FUTURE VIEWS IN DATABASE**, **FUTURE EXTERNAL TABLES IN DATABASE**, **FUTURE DYNAMIC TABLES IN DATABASE**, **FUTURE STREAMS IN DATABASE**
+Read-only access to query future objects.
+
+- --
+
+## Object creation
+
+In the setup guide, you'll be asked to create the following object(s) in your account.
+
+You'll need to execute [this script](https://raw.githubusercontent.com/monte-carlo-data/sna-artemis-agent/refs/heads/main/scripts/setup_app.sql), which will create the following objects:
+- `MCD_AGENT_ROLE`: role used to execute queries.
+- `MCD_AGENT_HELPER` database owned by `MCD_AGENT_ROLE`, which includes:
+    - `MCD_AGENT_EXECUTE_QUERY` stored procedure used to execute queries on behalf of `MCD_AGENT_ROLE`.
+    - This stored procedure is used to avoid a limitation with Snowflake Applications (not being able to grant `FUTURE GRANTS` to them).
+    - Accessing future objects and retain existing permissions on re-created objects is critical for Monte Carlo’s observability features.
+    - For more information, see [FAQ for the Snowflake Agent](https://docs.getmontecarlo.com/docs/sna-agent-deployment#faqs).
+
+Additionally, the following objects will be created in the application's database:
+- `CORE.MCD_AGENT_TOKEN`: Secret object used to store the authentication token to Monte Carlo platform.
+- `CORE.DATA_STORE`: Internal stage used to store data when the response exceeds a given threshold or for some Monte Carlo features like data sampling.
```

Comment:
You can link to the docs on sampling: https://docs.getmontecarlo.com/docs/data-sampling